### PR TITLE
Added Support for angular 1.4.1

### PR DESCRIPTION
**Pull request by kamthamc**
* Updated old npm libraries to new ones, fixed unit tests
* added support for fullcalendar 2.3 and angular 1.4
* Something in angular 1.3.0.rc-4 has broke the calendar(view will not update visible unless any view or date is clicked). For the calendar to work use to timeout to render the calendar.

### CODE REVIEW

File: src/calendar.js [hunk 1]
```diff
@@ -1,310 +1,360 @@
...
-                  }
-              };
-          };
-
-      var eventSerialId = 1;
-      // @return {String} fingerprint of the event object and its properties
-      this.eventFingerprint = function(e) {
-        if (!e._id) {
-          e._id = eventSerialId++;
+    .constant('uiCalendarConfig', {calendars: {}})
```
Comment (galuszkak) [inline src/calendar.js]: Can you remove that or this is on purpose? @kamthamc this make unreadable what are real changes in the code and what is only change of formatting... I propose changing formatting in seperate PR if it's necessary.

**Comment by ballerabdude** [comment]
There has been some changes merged in before this one which may have created a conflict. @kamthamc try to pull the latest.

**Comment by kamthamc** [comment]
@ballerabdude Reverted Formatting and fixed merge conflicts

**Comment by galuszkak** [comment]
Now it looks far more better to read!! Thanks @kamthamc :)

**Comment by galuszkak** [comment]
+1 for me. I maybe don't understand part with changing internal methods like destroy, or init. But if test works then I'm ok with that :+1: .

**Comment by kamthamc** [comment]
Angular has scope.$destroy method. So, just renamed to avoid confusion. Used same name convention for init.


Thanks and Regards
Kamatham Chaitanya Kumar

**Comment by ballerabdude** [comment]
if you try running grunt we get an angular not defined error and when I add angular to the global list it still fails. Can you take a look please?

**Comment by kamthamc** [comment]
Did you do do a bower install. The tests and demo are working which is isn't possible if angular is not defined


Thanks and Regards
Kamatham Chaitanya Kumar

**Pull request merged**